public/jira/rolledup-and-rolledback/rollup-and-rollback.ts

```diff
@@ -1,33 +1,50 @@
 import { rollbackIssues } from "../raw/rollback/rollback";
 import { DerivedIssue, deriveIssue } from "../derived/derive";
 import { normalizeIssue, NormalizeIssueConfig } from "../normalized/normalize";
-import { addRollupDates } from "../rollup/dates/dates";
-import { addWorkTypeDates } from "../rolledup/work-type/work-type";
-import { rollupBlockedStatusIssues } from "../rollup/blocked-status-issues/blocked-status-issues";
+import { addRollupDates, WithDateRollup } from "../rollup/dates/dates";
+import { addWorkTypeDates, WithWorkTypeRollups } from "../rolledup/work-type/work-type";
+import {
+  rollupBlockedStatusIssues,
+  WithBlockedStatuses,
+} from "../rollup/blocked-status-issues/blocked-status-issues";
 import { deriveReleases } from "../releases/derive";
 import { normalizeReleases } from "../releases/normalize";
-import { addPercentComplete } from "../rollup/percent-complete/percent-complete";
-import { addReportingHierarchy, IssueOrRelease } from "../rollup/rollup";
-import { rollupChildStatuses } from "../rollup/child-statuses/child-statuses";
-import { rollupWarningIssues } from "../rollup/warning-issues/warning-issues";
+import {
+  addPercentComplete,
+  WithPercentComplete,
+} from "../rollup/percent-complete/percent-complete";
+import { addReportingHierarchy, IssueOrRelease, WithReportingHierarchy } from "../rollup/rollup";
+import { rollupChildStatuses, WithChildStatuses } from "../rollup/child-statuses/child-statuses";
+import { rollupWarningIssues, WithWarningIssues } from "../rollup/warning-issues/warning-issues";
 import {
   addHistoricalAdjustedEstimatedTime,
   FEATURE_HISTORICALLY_ADJUSTED_ESTIMATES,
 } from "../rollup/historical-adjusted-estimated-time/historical-adjusted-estimated-time";
 import { JiraIssue, RollupLevelAndCalculation } from "../shared/types";
 
-type ExtractIssueOrReleaseType<T> = T extends IssueOrRelease<infer X> ? X : never;
-
-export type WithIssueLastPeriod<T> = {
+export type WithIssueLastPeriod<T = unknown> = {
   issueLastPeriod: IssueOrRelease<T>;
 };
 
+export type WithRollups = WithReportingHierarchy &
```

Comment:
not a blocker but is this exported type actually used anywhere else? i can't find it imported anywhere in the PR

Reply:
it's used on line 47 of this file, I think I was using it somewhere else in a previous version, and I left it exported cuz it seemed like something we were likely to need elsewhere

Reply:
It seems like the kind of type that might be better served when pulled out to a common types file, _if_ you're using it externally.